homeassistant/components/tod/binary_sensor.py

```diff
@@ -122,9 +122,10 @@ def __init__(
         """Init the ToD Sensor..."""
         self._attr_unique_id = unique_id
         self._attr_name = name
-        self._time_before: datetime | None = None
-        self._time_after: datetime | None = None
-        self._next_update: datetime | None = None
+        # Not correctly typed so we don't need to assert the typing everywhere
+        self._time_before: datetime = None  # type: ignore[assignment]
+        self._time_after: datetime = None  # type: ignore[assignment]
+        self._next_update: datetime = None  # type: ignore[assignment]
```

Comment:
Can they be moved to class attributes instead?
```python
    _time_before: datetime
    _time_after: datetime
    _next_update: datetime
```

Reply:
Yes, better